File: server/src/knowledge/ingestors/textIngestor.js
import { BaseIngestor } from '../baseIngestor.js';
import { parseDocument, chunkText } from '../../rag.js';

// Real implementation for txt/markdown — same chunker, no page concept.
export class TextIngestor extends BaseIngestor {
  constructor() {
    super('document');
  }

  async ingest(source) {
    const text = await parseDocument(source.filePath, source.mimeType || 'text/plain');
    const chunks = chunkText(text);
    return chunks.map(content => ({ content, metadata: {} }));
  }
}

File: server/app/knowledge/ingestors/text_ingestor.py
from ..base_ingestor import BaseIngestor
from ...rag import parse_document, chunk_text


class TextIngestor(BaseIngestor):
    """Real implementation for txt/markdown — same chunker, no page concept."""

    def __init__(self):
        super().__init__("document")

    async def ingest(self, source: dict) -> list[dict]:
        parsed = await parse_document(source["filePath"], source.get("mimeType", "text/plain"))
        chunks = chunk_text(parsed["text"])
        return [{"content": c, "metadata": {}} for c in chunks]
